resource-providers/src/main/java/io/opentelemetry/resource/spring/SpringBootServiceNameGuesser.java

```diff
@@ -0,0 +1,255 @@
+  }
+
+  @Nullable
+  private String findByCurrentDirectoryApplicationProperties() {
+    String result = null;
+    try (InputStream in = system.openFile("application.properties")) {
+      result = getAppNamePropertyFromStream(in);
+    } catch (Exception e) {
+      // expected to fail sometimes
+    }
+    logger.debug("Checking application.properties in current dir: " + result);
+    return result;
+  }
+
+  @Nullable
+  private String findByClasspathApplicationYaml() {
+    String result =
+        loadFromClasspath("application.yml", SpringBootServiceNameGuesser::parseNameFromYaml);
+    logger.debug("Checking application.yml in classpath: " + result);
+    return result;
+  }
+
+  @Nullable
+  private String findByCurrentDirectoryApplicationYaml() {
+    String result = null;
+    try (InputStream in = system.openFile("application.yml")) {
+      result = parseNameFromYaml(in);
+    } catch (Exception e) {
+      // expected to fail sometimes
+    }
+    logger.debug("Checking application.yml in current dir: " + result);
+    return result;
+  }
+
+  @Nullable
+  @SuppressWarnings("unchecked")
+  private static String parseNameFromYaml(InputStream in) {
+    Yaml yaml = new Yaml();
+    try {
+      Map<String, Object> data = yaml.load(in);
+      Map<String, Map<String, Object>> spring =
+          (Map<String, Map<String, Object>>) data.get("spring");
+      if (spring != null) {
+        Map<String, Object> app = spring.get("application");
+        if (app != null) {
+          Object name = app.get("name");
+          if (name != null) {
+            return (String) name;
+          }
+        }
+      }
+    } catch (RuntimeException e) {
+      // expected to fail sometimes
+    }
+    return null;
+  }
+
+  @Nullable
+  private String findByCommandlineArgument() {
+    String result = attemptProcessHandleReflection();
+    if (result == null) {
+      String javaCommand = system.getProperty("sun.java.command");
+      result = parseNameFromCommandLine(javaCommand);
+    }
+    logger.debug("Checking application commandline args: " + result);
+    return result;
+  }
+
+  @Nullable
+  private String attemptProcessHandleReflection() {
+    try {
+      String commandLine = system.attemptGetCommandLineViaReflection();
+      return parseNameFromCommandLine(commandLine);
+    } catch (Exception e) {
+      return null;
+    }
+  }
+
+  private static String parseNameFromCommandLine(String commandLine) {
+    Matcher matcher = COMMANDLINE_PATTERN.matcher(commandLine);
+    boolean ignored = matcher.find(); // Required before group()
+    return matcher.group(1);
+  }
+
+  @Nullable
+  private String readNameFromAppProperties() {
+    return loadFromClasspath(
+        "application.properties", SpringBootServiceNameGuesser::getAppNamePropertyFromStream);
+  }
+
+  @Nullable
+  private static String getAppNamePropertyFromStream(InputStream in) {
+    Properties properties = new Properties();
+    try {
+      properties.load(in);
```

Comment:
would it make sense to add a comment that `Properties.load` use `ISO-8859-1` encoding which is also what spring uses by default for properties files?

Reply:
Sure, since we chatted about encodings, I think that makes sense. 👍🏻